pyigm/clustering/xcorr_utils.py

```diff
@@ -322,7 +386,175 @@ def spline_sensitivity(galreal, Nmin=20, magmin=17., magmax=26., delta_mag=0.5,
     return magbins, VALS, SPL
 
 
-def random_gal(galreal, Nrand, magbins, SPL):
+def random_abs_zmnx(absreal, Nrand, zmnx, wrest, dv_Galactic=100.):
+    """From a real absorber catalog it creates a random catalog.
+    Absorbers are simply placed randomly between redshift limits
+    zmnx[0] and zmnx[1]
+
+    Parameters
+    ----------
+    absreal
+    Nrand
+    zmnx
+
+    Returns
+    -------
+
+    """
+
+    absrand = absreal.repeat(Nrand)
+
+    randz = np.random.uniform(low=zmnx[0], high=zmnx[1], size=2*len(absrand))  # Buffer to reject Galaxy
+
+    # Avoid Galactic
+    galactic = mask_galactic()
+    Galz = galactic/wrest - 1.
+    z_Gal = np.outer(np.ones_like(randz), Galz)
+    # Diff
+    diff = z_Gal - np.outer(randz, np.ones_like(Galz))
+    mdiff = np.amin(np.abs(diff), axis=1)
+    # Good ones
+    gdz = mdiff > (dv_Galactic/Ckms) / (randz+1)
+    gdi = np.where(gdz)[0]
+    absrand.ZABS = randz[gdi[0:len(absrand)]]
+    # Return
+    return absrand
+
+
+def random_abs_W(absreal, Nrand, wa, fl, er, sl=3., R=20000, FWHM=10., ion='HI'):
```

Comment:
Change name to `random_abs_Wr` to avoid confusion with the W the 2PCF.

Reply:
ok